### PR TITLE
cleanup: attempt to clean up lookupOnlineAccountData

### PR DESCRIPTION
**Pull request by brianolson**
## Summary

Attempt to clean up lookupOnlineAccountData() in response to pull request feedback.

## Test Plan

Still passes unit tests.

### CODE REVIEW

File: ledger/acctonline.go [hunk 1]
```diff
@@ -210,14 +213,14 @@ func (ao *onlineAccounts) close() {
...
+	ao.newBlockInner(blk, delta)
 	ao.accountsMu.Unlock()
 	ao.accountsReadCond.Broadcast()
 }
 
-// newBlockImpl is the accountUpdates implementation of the ledgerTracker interface. This is the "internal" facing function
+// newBlockInner is the accountUpdates implementation of the ledgerTracker interface. This is the "internal" facing function
 // which assumes that no lock need to be taken.
-func (ao *onlineAccounts) newBlockImpl(blk bookkeeping.Block, delta ledgercore.StateDelta) {
+func (ao *onlineAccounts) newBlockInner(blk bookkeeping.Block, delta ledgercore.StateDelta) {
```
[the remark this answers is not in the document]
Comment (cce) [reply ledger/acctonline.go:223]: we also have a lot of methods that end with Impl throughout the codebase

File: ledger/acctonline.go [hunk 2]
```diff
@@ -633,7 +635,8 @@ func (ao *onlineAccounts) lookupOnlineAccountData(rnd basics.Round, addr basics.
 				// Check if this is the most recent round, in which case, we can
 				// use a cache of the most recent account state.
 				if offset == uint64(len(ao.deltas)) {
-					return macct.data.OnlineAccountData(rewardsProto, rewardsLevel), nil
+					out = macct.data.OnlineAccountData(rewardsProto, rewardsLevel)
```
Comment (cce) [inline ledger/acctonline.go:638]: It feels more complex to me to have set named output variables from from inside a closure, inside a helper function, versus just being able to return them directly to the caller as before ...

File: ledger/acctonline.go [hunk 3]
```diff
@@ -644,81 +647,97 @@ func (ao *onlineAccounts) lookupOnlineAccountData(rnd basics.Round, addr basics.
...
+	//   2. after commitRound but before postCommit (currentDbRound >= ao.cachedDBRoundOnline && currentDeltaLen == len(ao.deltas)) => OK
+	//      the cache gets populated and postCommit updates the new entry
+	//   3. after commitRound and after postCommit => problem
+	//      postCommit does not add a new entry, but the cache that would get constructed would miss the latest entry, retry
+	// In order to resolve this lookupOnlineHistory returns dbRound value (as validThrough) and determine what happened
+	// So handle cases 3.1 and 3.2 here, and 3.3 below
+	out, err = ao.lookupOnlineAccountDataHandleData(addr, currentDbRound, currentDeltaLen, &persistedData, persistedDataHistory, validThrough, rewardsProto, rewardsLevel)
+	if err == errNotDoneYet {
+		// case 3.3: retry (for loop iterates and queries again)
+		goto slowDbRetry
```
Comment (cce) [inline ledger/acctonline.go:705]: why a goto vs a for loop? They are pretty similar ... is there a reason for one over the other?

File: ledger/acctonline.go [hunk 4]
```diff
@@ -644,81 +647,97 @@ func (ao *onlineAccounts) lookupOnlineAccountData(rnd basics.Round, addr basics.
...
+	}
+	return
```
Comment (cce) [inline ledger/acctonline.go:707]: I really dislike naked returns like this one in long functions — IMO it makes things hard to follow once you are a page of text or so below the beginning of the function. https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters

**Comment by algorandskiy** [comment]
I suggest to postpone it until after the release
